src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt

```diff
@@ -24,29 +22,123 @@ class ThemePatch : ResourcePatch {
 
-        context.xmlEditor["res/values/colors.xml"].use { editor ->
-            val resourcesNode = editor.file.getElementsByTagName("resources").item(0) as Element
+        val valuesPaths: List<String> = listOf(
+            "values", // Android 11 (and below)
+            "values-v31" // Android 12 (and above)
+        )
+
+        // edit the resource color file to change the theme color
+        val colorsPath = "res/$valuesPaths[0]/colors.xml"
+
+        context.xmlEditor[colorsPath].use { editor ->
+            val childNodes = (editor.file.getElementsByTagName("resources").item(0) as Element).childNodes
 
-            for (i in 0 until resourcesNode.childNodes.length) {
-                val node = resourcesNode.childNodes.item(i) as? Element ?: continue
+            for (i in 0 until childNodes.length) {
+                val node = childNodes.item(i) as? Element ?: continue
 
                 node.textContent = when (node.getAttribute("name")) {
                     "yt_black0", "yt_black1", "yt_black1_opacity95", "yt_black1_opacity98", "yt_black2", "yt_black3",
                     "yt_black4", "yt_status_bar_background_dark", "material_grey_850" -> darkThemeBackgroundColor
 
-                    "yt_white1", "yt_white1_opacity95", "yt_white1_opacity98", "yt_white2", "yt_white3", "yt_white4",
-                     -> lightThemeBackgroundColor
+                    "yt_white1", "yt_white1_opacity95", "yt_white1_opacity98", "yt_white2", "yt_white3", "yt_white4"
+                    -> lightThemeBackgroundColor
 
                     else -> continue
                 }
             }
         }
 
-        // copies the resource file to change the splash screen color
-        context.copyResources("theme",
-            ResourceUtils.ResourceGroup("values-night-v31", "styles.xml")
+        // edit the resource files to change the splash screen color
+        val attrsPath = "res/$valuesPaths[0]/attrs.xml"
+        val drawablePaths: List<String> = listOf(
+            "drawable",
+            "drawable-anydpi",
+            "drawable-hdpi",
+            "drawable-ldrtl",
+            "drawable-ldrtl-night",
+            "drawable-ldrtl-xxhdpi",
+            "drawable-mdpi",
+            "drawable-night",
+            "drawable-nodpi",
+            "drawable-sw600dp",
+            "drawable-sw600dp-xhdpi",
+            "drawable-sw720dp-xhdpi",
+            "drawable-watch",
+            "drawable-xhdpi",
+            "drawable-xxhdpi",
+            "drawable-xxxhdpi"
         )
 
+        context.xmlEditor[attrsPath].use { editor ->
+            val file = editor.file
+
+            val resourceNode = file.getElementsByTagName("resources").item(0) as Element
+
+            resourceNode.appendChild(
+                file.createElement("attr").apply {
+                    setAttribute("format", "reference")
+                    setAttribute("name", "splashScreenColor")
+                }
+            )
```

Comment:
Please document this, it is unknown what this code does or why it is required.

Reply:
This code is useful to create a a new xml line inside ```attr.xml``` (with all required attributes of other lines inside the file), named ```splashScreenColor```

Reply:
It is unnecessary to explain that it creates a new line in the `attr.xml` file. Explain why it is required and what for purpose.

Reply:
> It is unnecessary to explain that it creates a new line in the `attr.xml` file. Explain why it is required and what for purpose.

I'll add a comment inside the patch then.